src/a01/models/task_collection.py

```diff
@@ -1,77 +0,0 @@
-import json
-from collections import defaultdict
-from typing import List, Tuple, Generator
-
-import colorama
-from requests import HTTPError
-
-from a01.common import get_logger, A01Config
-from a01.communication import session
-from a01.models import Task
-
-
-class TaskCollection(object):
-    logger = get_logger('TaskCollection')
-
-    def __init__(self, tasks: List[Task], run_id: str) -> None:
-        self.tasks = sorted(tasks, key=lambda t: t.identifier)
-        self.run_name = run_id
-
-    def get_failed_tasks(self) -> Generator[Task, None, None]:
-        for task in self.tasks:
-            if task.result == 'Passed' or task.status == 'initialized':
-                continue
-            yield task
-
-    def post(self) -> None:
-        try:
-            payload = [t.to_dict() for t in self.tasks]
-            session.post(f'{self.endpoint_uri()}/run/{self.run_name}/tasks', json=payload).raise_for_status()
-        except HTTPError:
-            self.logger.debug('HttpError', exc_info=True)
-            raise ValueError('Failed to create tasks in the task store.')
-
```

Comment:
This went to the dispatcher IIRC?

Reply:
They don't need. The `requests` is going to be replaced by `aiohttp`. And the logic of making HTTP requests will be decoupled from the model.